crates/ethrpc/src/http.rs

```diff
@@ -72,19 +72,33 @@ async fn execute_rpc<T: DeserializeOwned>(
 ) -> Result<T, Web3Error> {
     let body = serde_json::to_string(&request)?;
     tracing::trace!(name = %inner.name, %id, %body, "executing request");
-    let mut request = client
+    let mut request_builder = client
         .post(inner.url.clone())
         .header(header::CONTENT_TYPE, "application/json")
         .header("X-RPC-REQUEST-ID", id.to_string())
         .body(body);
-    if let Some(request_id) = observe::request_id::get_task_local_storage() {
-        request = request.header("X-REQUEST-ID", request_id);
+    match request {
+        Request::Single(Call::MethodCall(method)) => {
+            if let Some(request_id) = observe::request_id::get_task_local_storage() {
+                request_builder = request_builder.header("X-REQUEST-ID", request_id);
+            }
+            request_builder = request_builder.header("X-RPC-METHOD", method.method.clone());
+        }
+        Request::Batch(_) => {
+            if let Some(metadata) = observe::request_id::get_task_local_storage() {
+                request_builder = request_builder.header("X-RPC-METADATA", metadata);
```

Comment:
```suggestion
                request_builder = request_builder.header("X-RPC-BATCH-METADATA", metadata);
```

to make it obvious that this is related to batch requests

Reply:
Decided to keep a single header with a unified format.